build/tools/verifyVendor.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+###
+#Copyright 2019 The KubeEdge Authors.
+#
+#Licensed under the Apache License, Version 2.0 (the "License");
+#you may not use this file except in compliance with the License.
+#You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#Unless required by applicable law or agreed to in writing, software
+#distributed under the License is distributed on an "AS IS" BASIS,
+#WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#See the License for the specific language governing permissions and
+#limitations under the License.
+###
+
+
+function checkModified {
+	modified=$( git status --short 2>/dev/null | grep -e "^.M" | wc -l)
+
+	if [ ${modified} -eq 0 ]; then
+		echo 0
+		return
+	fi
+
+	echo 2
+}
+ 
+go mod vendor
+ret=$(checkModified)
+if [ ${ret} -eq 0 ]; then
+	echo "SUCCESS: vendor is up to date"
+else
+	echo  "FAILED: vendor needs an update; The diff is:"
+	git diff
+	exit 1
+fi
+
+go get -u=patch
```

Comment:
why use `go get -u=patch` ?  It will instructs get to update dependencies
to use newer patch releases when available.

Is it necessary ? @subpathdev @sids-b

Reply:
> It will instructs get to update dependencies to use newer patch releases when available.
Yes you are right

> Is it necessary ?
You have to decide what we want to do. If we want to check if one of our dependencies has been patched then yes it is necessary. If we only check if the vendor has been build after changing the go.mod or go.sum file it is not necessary.

I think we should check if one of our dependencies has been updated. We should include  those patches to not include security issues in our programs. (the minor version will be checked, in an example version string v1.x.y the y is the part who will checked by go get -u=patch.)

Reply:
@subpathdev ,  that sounds good, but  this will force us to use the latest patch version of the dependency.  This may lead to compatibility issues。 kubernetes also does not use this feature。

Reply:
I think we should discuss this on an separate pr or?